src/script/interpreter.h

```diff
@@ -98,7 +98,20 @@ struct PrecomputedTransactionData
     /** Precomputed transaction parts. */
     std::unique_ptr<PrecomputedTxParts, decltype(&zcash_transaction_precomputed_free)> preTx;
 
-    PrecomputedTransactionData(const CTransaction& tx);
+    PrecomputedTransactionData(
+        const CTransaction& tx,
+        const std::vector<CTxOut>& allPrevOutputs);
+
+    PrecomputedTransactionData(
+        const CTransaction& tx,
+        const unsigned char* allPrevOutputs,
+        size_t allPrevOutputsLen);
+
+private:
+    void SetPrecomputed(
+        const CTransaction& tx,
+        const unsigned char* allPrevOutputs,
+        size_t allPrevOutputsLen);
```

Comment:
Suggestion: replace this with a static method that returns `std::optional<PrecomputedTransactionData>`, and add similar static method to replace the `PrecomputedTransactionData` that accepts a `std::vector<CTxOut>` thatdelegates to this one and then unwraps the result, since that should be infallible.

I guess that this would make it impossible to use `emplace_back` for a bunch of places where you're using this construction. I still do think it would be better to replace the second constructor with a static factory method though.

Reply:
I took the current approach in order to minimise this part of the diff. Once we have the `zcash_script` usage of these constructors in place, I think we could look at a targeted refactor of this part, but I'd prefer not to alter it in this PR.